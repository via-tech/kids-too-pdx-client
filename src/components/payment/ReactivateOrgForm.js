import React from 'react';
import PropTypes from 'prop-types';
import UserInfo from '../users/UserInfo';
import PaymentInfo from './PaymentInfo';
import styles from '../users/SignInForm.css';

function ReactivateOrgForm(props) {
  const { handleSubmit, user, error } = props;
  
  return (
    <form className={styles.SignIn} onSubmit={handleSubmit.bind(null, user)}>
      {error && <p className={styles.error}>{error}</p>}

      <UserInfo {...props} />

      <PaymentInfo {...props} />

      <button type="submit">Activate Account</button>
    </form>
  );
}

ReactivateOrgForm.propTypes = {
  handleSubmit: PropTypes.func,
  user: PropTypes.object,
  error: PropTypes.string
};

export default ReactivateOrgForm;
