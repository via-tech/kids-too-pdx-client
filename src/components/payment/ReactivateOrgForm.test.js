import React from 'react';
import renderer from 'react-test-renderer';
import ReactivateOrgForm from './ReactivateOrgForm';

describe('Reactivation Form', () => {
  it('snapshots ReactivateOrgForm', () => {
    const props = {
      handleChange: jest.fn(),
      handleSubmit: jest.fn(),
      user: {
        username: 'theOrg@org.com',
        password: 'pass'
      },
      subFee: '900'
    };

    const tree = renderer.create(
      <ReactivateOrgForm {...props} />
    );

    expect(tree).toMatchSnapshot();
  });

  it('snapshots ReactivateOrgForm with error', () => {
    const props = {
      handleChange: jest.fn(),
      handleSubmit: jest.fn(),
      user: {
        username: 'theOrg@org.com',
        password: 'pass'
      },
      subFee: '900',
      error: 'User already exists'
    };

    const tree = renderer.create(
      <ReactivateOrgForm {...props} />
    );

    expect(tree).toMatchSnapshot();
  });
});
