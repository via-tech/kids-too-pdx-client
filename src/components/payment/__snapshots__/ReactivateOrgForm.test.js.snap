// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reactivation Form snapshots ReactivateOrgForm 1`] = `
<form
  className="SignIn"
  onSubmit={[Function]}
>
  <fieldset>
    <legend>
      User Info
    </legend>
    <label>
      Username or Email:
      <input
        name="username"
        onChange={[MockFunction]}
        required={true}
        type="text"
      />
    </label>
    <label>
      Password:
      <input
        name="password"
        onChange={[MockFunction]}
        required={true}
        type="password"
      />
    </label>
  </fieldset>
  <fieldset>
    <legend>
      Payment Info
    </legend>
    <label>
      Card Number
      <span
        className="required"
      >
        *
      </span>
      <input
        maxLength="19"
        name="cardNumber"
        onChange={[MockFunction]}
        pattern="[0-9]*"
        placeholder="Card Number"
        required={true}
        type="text"
      />
    </label>
    <label>
      Security Code
      <span
        className="required"
      >
        *
      </span>
      <input
        maxLength="3"
        name="securityCode"
        onChange={[MockFunction]}
        required={true}
        type="text"
      />
    </label>
    <label>
      Name
      <span
        className="required"
      >
        *
      </span>
      <input
        name="cardName"
        onChange={[MockFunction]}
        placeholder="Card Name"
        required={true}
        type="text"
      />
    </label>
    <label>
      Expiration Date
      <span
        className="required"
      >
        *
      </span>
      <input
        maxLength="2"
        name="expMonth"
        onChange={[MockFunction]}
        pattern="[0-9]*"
        placeholder="mm"
        required={true}
        type="text"
      />
      <input
        maxLength="4"
        minLength="4"
        name="expYear"
        onChange={[MockFunction]}
        pattern="[0-9]*"
        placeholder="yyyy"
        required={true}
        type="text"
      />
    </label>
    <label>
      Method
      <span
        className="required"
      >
        *
      </span>
      <input
        name="method"
        onChange={[MockFunction]}
        required={true}
        type="radio"
        value="visa"
      />
      Visa
      <input
        name="method"
        onChange={[MockFunction]}
        required={true}
        type="radio"
        value="master"
      />
      Master
    </label>
    <label>
      Billing Address 
      <span
        className="required"
      >
        *
      </span>
      <label>
        Street:
        <input
          name="billStreet"
          onChange={[MockFunction]}
          type="text"
        />
      </label>
      <label>
        City:
        <input
          name="billCity"
          onChange={[MockFunction]}
          type="text"
        />
      </label>
      <label>
        State
        <input
          name="state"
          onChange={[MockFunction]}
          required={true}
          type="text"
        />
      </label>
      <label>
        Zip Code:
        <input
          name="billZipcode"
          onChange={[MockFunction]}
          pattern="[0-9]*"
          type="text"
        />
      </label>
    </label>
  </fieldset>
  <h2>
    Total: $
    900
  </h2>
  <button
    type="submit"
  >
    Activate Account
  </button>
</form>
`;

exports[`Reactivation Form snapshots ReactivateOrgForm with error 1`] = `
<form
  className="SignIn"
  onSubmit={[Function]}
>
  <p
    className="error"
  >
    User already exists
  </p>
  <fieldset>
    <legend>
      User Info
    </legend>
    <label>
      Username or Email:
      <input
        name="username"
        onChange={[MockFunction]}
        required={true}
        type="text"
      />
    </label>
    <label>
      Password:
      <input
        name="password"
        onChange={[MockFunction]}
        required={true}
        type="password"
      />
    </label>
  </fieldset>
  <fieldset>
    <legend>
      Payment Info
    </legend>
    <label>
      Card Number
      <span
        className="required"
      >
        *
      </span>
      <input
        maxLength="19"
        name="cardNumber"
        onChange={[MockFunction]}
        pattern="[0-9]*"
        placeholder="Card Number"
        required={true}
        type="text"
      />
    </label>
    <label>
      Security Code
      <span
        className="required"
      >
        *
      </span>
      <input
        maxLength="3"
        name="securityCode"
        onChange={[MockFunction]}
        required={true}
        type="text"
      />
    </label>
    <label>
      Name
      <span
        className="required"
      >
        *
      </span>
      <input
        name="cardName"
        onChange={[MockFunction]}
        placeholder="Card Name"
        required={true}
        type="text"
      />
    </label>
    <label>
      Expiration Date
      <span
        className="required"
      >
        *
      </span>
      <input
        maxLength="2"
        name="expMonth"
        onChange={[MockFunction]}
        pattern="[0-9]*"
        placeholder="mm"
        required={true}
        type="text"
      />
      <input
        maxLength="4"
        minLength="4"
        name="expYear"
        onChange={[MockFunction]}
        pattern="[0-9]*"
        placeholder="yyyy"
        required={true}
        type="text"
      />
    </label>
    <label>
      Method
      <span
        className="required"
      >
        *
      </span>
      <input
        name="method"
        onChange={[MockFunction]}
        required={true}
        type="radio"
        value="visa"
      />
      Visa
      <input
        name="method"
        onChange={[MockFunction]}
        required={true}
        type="radio"
        value="master"
      />
      Master
    </label>
    <label>
      Billing Address 
      <span
        className="required"
      >
        *
      </span>
      <label>
        Street:
        <input
          name="billStreet"
          onChange={[MockFunction]}
          type="text"
        />
      </label>
      <label>
        City:
        <input
          name="billCity"
          onChange={[MockFunction]}
          type="text"
        />
      </label>
      <label>
        State
        <input
          name="state"
          onChange={[MockFunction]}
          required={true}
          type="text"
        />
      </label>
      <label>
        Zip Code:
        <input
          name="billZipcode"
          onChange={[MockFunction]}
          pattern="[0-9]*"
          type="text"
        />
      </label>
    </label>
  </fieldset>
  <h2>
    Total: $
    900
  </h2>
  <button
    type="submit"
  >
    Activate Account
  </button>
</form>
`;
